torch/csrc/jit/passes/autocast.cpp

```diff
@@ -216,6 +216,39 @@ void castInputsToWidestType(Node* node, const AutocastContext& context) {
   }
 }
 
+// Users can call torch.is_autocast_enabled() or is_autocast_cpu_enabled() to
+// determine whether autocasting is enabled. With JIT-scripted functions, we
+// actually need to return true if eager autocast OR jit autocast are enabled.
+//
+// In the case where JIT autocast is enabled, we replace
+//    %x : bool = aten::is_autocast_enabled()
+// with a constant "True".
+//
+// More context on eager vs JIT autocasting:
+//
+// Autocasting actually has two settings: eager autocasting, and JIT
+// autocasting. Eager autocasting is the thread-local setting that turns on
+// the relevant bit in the dispatcher settings. JIT autocasting is the pass
+// implemented in this file, which makes changes to the graph to insert casting
+// ops in order to achieve the same behavior as eager autocasting.
+//
+// If eager autocasting is enabled at the time when a JIT-scripted function is
+// invoked, then autocasting will occur regardless of what the JIT-autocasting
+// settings are.
+void updateAutocastEnabledCheck(Node* node, bool is_jit_enabled) {
+  if (!is_jit_enabled) {
+    return;
+  }
+
+  auto graph = node->owningGraph();
+
+  WithInsertPoint insert_point(node);
+
+  Value* true_constant = graph->insertConstant(IValue(true));
+  node->output()->replaceAllUsesWith(true_constant);
+  node->destroy();
```

Comment:
@eellison would there be any concern here that this makes debugging hard? This will entirely remove the `aten::is_autocast_enabled` node & replace it with a constant

Reply:
Don't think it's that much different than the pass that specializes the value.